node/external/blockAPI/shardBlock.go

```diff
@@ -83,6 +87,66 @@ func (sbp *shardAPIBlockProcessor) GetBlockByRound(round uint64, options api.Blo
 	return sbp.convertShardBlockBytesToAPIBlock(headerHash, blockBytes, options)
 }
 
+// GetAlteredAccountsForBlock will return the altered accounts for the desired shard block
+func (sbp *shardAPIBlockProcessor) GetAlteredAccountsForBlock(options api.GetAlteredAccountsForBlockOptions) (*common.AlteredAccountsForBlockAPIResponse, error) {
+	headerHash, blockBytes, err := sbp.getHashAndBlockBytesFromStorer(options.GetBlockParameters)
+	if err != nil {
+		return nil, err
+	}
+
+	apiBlock, err := sbp.convertShardBlockBytesToAPIBlock(headerHash, blockBytes, api.BlockQueryOptions{WithTransactions: true, WithLogs: true})
+	if err != nil {
+		return nil, err
+	}
+
+	txPool := sbp.apiBlockToTxsPool(apiBlock)
+
+	rootHash, err := hex.DecodeString(apiBlock.StateRootHash)
+	if err != nil {
+		return nil, err
+	}
+
+	alteredAccountsOptions := shared.AlteredAccountsOptions{
+		WithCustomAccountsRepository: true,
+		AccountsRepository:           sbp.accountsRepository,
+		// TODO: AccountQueryOptions could be used like options.WithBlockRootHash(..) instead of thinking what to provide
+		AccountQueryOptions: api.AccountQueryOptions{
+			BlockRootHash: rootHash,
+		},
+	}
+
+	alteredAccounts, err := sbp.alteredAccountsProvider.ExtractAlteredAccountsFromPool(txPool, alteredAccountsOptions)
```

Comment:
Not sure why.

Reply:
Now I think I know - for code reuse? :)

Reply:
Yes, it is for code reuse and not to change much logic that is currently implemented. Your suggestion implies a refactoring on both shard/meta api block processor (return regular txs instead of the api block) and altered accounts provider (which currently relies on that outport pool